[Bug]: Freshly installed Meltano unusable due to werkzeug update

### Meltano Version

2.3.0

### Python Version

NA

### Bug scope

CLI (options, error messages, logging, etc.)

### Operating System

Any

### Description

On a freshly installed Meltano, any CLI command fails due to an import error in the `flask-login` dependency.

### Code

```python
$ meltano --version
Traceback (most recent call last):
  File "/opt/pipx_bin/meltano", line 5, in <module>
    from meltano.cli import main
  File "/opt/pipx/venvs/meltano/lib/python3.8/site-packages/meltano/cli/__init__.py", line 20, in <module>
    from . import (  # isort:skip # noqa: F401, WPS235
  File "/opt/pipx/venvs/meltano/lib/python3.8/site-packages/meltano/cli/user.py", line 5, in <module>
    from meltano.api.app import create_app
  File "/opt/pipx/venvs/meltano/lib/python3.8/site-packages/meltano/api/app.py", line 13, in <module>
    from meltano.api.security.auth import HTTP_READONLY_CODE
  File "/opt/pipx/venvs/meltano/lib/python3.8/site-packages/meltano/api/security/__init__.py", line 1, in <module>
    from flask_login import user_logged_in
  File "/opt/pipx/venvs/meltano/lib/python3.8/site-packages/flask_login/__init__.py", line 12, in <module>
    from .login_manager import LoginManager
  File "/opt/pipx/venvs/meltano/lib/python3.8/site-packages/flask_login/login_manager.py", line 35, in <module>
    from .utils import _create_identifier
  File "/opt/pipx/venvs/meltano/lib/python3.8/site-packages/flask_login/utils.py", line 14, in <module>
    from werkzeug.routing import parse_rule
ImportError: cannot import name 'parse_rule' from 'werkzeug.routing' (/opt/pipx/venvs/meltano/lib/python3.8/site-packages/werkzeug/routing/__init__.py)
Error: Process completed with exit code 1.
```

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -61,7 +61,7 @@ sqlalchemy = "^1.3.19"
 structlog = "^21.2.0"
 tzlocal = "^4.2.0"
 uvicorn = {extras = ["standard"], version = "^0.17.6"}
-werkzeug = "<=2.1.3"
+werkzeug = ">=2.1,<=2.1.3"
 
 [tool.poetry.extras]
 infra = ["ansible"]

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -61,7 +61,7 @@ sqlalchemy = "^1.3.19"
 structlog = "^21.2.0"
 tzlocal = "^4.2.0"
 uvicorn = {extras = ["standard"], version = "^0.17.6"}
-werkzeug = "^2.1"
+werkzeug = "<=2.1.3"
 
 [tool.poetry.extras]
 infra = ["ansible"]

